### PR TITLE
added egnaddrs binary

### PR DESCRIPTION
**Pull request by samlaf**
Implements #45 .

### CODE REVIEW

File: cmd/egnaddrs/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```bash
+   go install
+```
+
+### Usage
+
+Currently egnaddrs only supports deriving contract addresses starting from the registry-coordinator address. It then prints the following datastructure:
+
+```bash
+$$$ egnaddrs --registry-coordinator 0x9E545E3C0baAB3E08CdfD552C960A1050f373042
```
Comment (shrimalmadhur) [inline cmd/egnaddrs/README.md:21]: can we also add one where we can input AVS service manager contracts and get the rest of them.

Comment (samlaf) [reply cmd/egnaddrs/README.md:21]: Done [7a60f64](https://github.com/Layr-Labs/eigensdk-go/pull/50/commits/7a60f64b9ecda1723dba68c663c2e5e144bacc52)

Also added tests in [6b29849](https://github.com/Layr-Labs/eigensdk-go/pull/50/commits/6b29849684b84bb421658429f69d7afec2033bed), please have a look

File: cmd/egnaddrs/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+#### Install from local source
+If you have git cloned eigensdk-go to your machine, navigate to [cmd/egnaddrs](.) directory and run
+```bash
+   go install
+```
+
+### Usage
+
+Currently egnaddrs only supports deriving contract addresses starting from the registry-coordinator address. It then prints the following datastructure:
```
Comment (shrimalmadhur) [inline cmd/egnaddrs/README.md:18]: can you update the readme to reflect that we can use service manager too.

Comment (samlaf) [reply cmd/egnaddrs/README.md]: done

**Review by shrimalmadhur — APPROVED** [review]

**Pull request merged**